<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Detection of Rare Aberrantly Methylated Regions in Array and NGS Data • ramr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Detection of Rare Aberrantly Methylated Regions in Array and NGS Data">
<meta property="og:description" content="ramr is an R package for detection of low-frequency aberrant
    methylation events in large datasets obtained by methylation profiling
    using array or high-throughput bisulfite sequencing. In addition, package
    provides functions to visualize found aberrantly methylated regions (AMRs),
    to generate sets of all possible regions to be used as reference sets for
    enrichment analysis, and to generate biologically relevant test datasets for
    performance evaluation of AMR/DMR search algorithms.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">ramr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/ramr.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/BBCG/ramr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="ramr" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#ramr" class="anchor" aria-hidden="true"></a>ramr</h1></div>

</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor" aria-hidden="true"></a>Introduction</h1>
<p><em><code>ramr</code></em> is an R package for detection of low-frequency aberrant methylation events in large datasets obtained by methylation profiling using array or high-throughput bisulfite sequencing. In addition, package provides functions to visualize found aberrantly methylated regions (AMRs), to generate sets of all possible regions to be used as reference sets for enrichment analysis, and to generate biologically relevant test datasets for performance evaluation of AMR/DMR search algorithms.</p>
<p>This readme contains condensed info on <em><code>ramr</code></em> usage. For more, please check function-specific help pages and vignettes within the R environment or at <a href="https://bbcg.github.io/ramr/articles/ramr.html" class="external-link">GitHub pages</a>.</p>
<div id="current-features" class="section level2">
<h2 class="hasAnchor">
<a href="#current-features" class="anchor" aria-hidden="true"></a>Current Features</h2>
<ul>
<li>Identification of aberrantly methylated regions (AMRs)</li>
<li>AMR visualization</li>
<li>Generation of reference sets for third-party analyses (e.g. enrichment)</li>
<li>Generation of test datasets for performance evaluation of algorithms for search of differentially (DMR) or aberrantly (AMR) methylated regions</li>
</ul>
<hr>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor" aria-hidden="true"></a>Installation</h2>
<div id="install-via-bioconductor" class="section level3">
<h3 class="hasAnchor">
<a href="#install-via-bioconductor" class="anchor" aria-hidden="true"></a>install via Bioconductor</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ramr"</span><span class="op">)</span></code></pre></div>
</div>
<div id="install-the-latest-version-via-install_github" class="section level3">
<h3 class="hasAnchor">
<a href="#install-the-latest-version-via-install_github" class="anchor" aria-hidden="true"></a>Install the latest version via install_github</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span>
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BBCG/ramr"</span>, build_vignettes<span class="op">=</span><span class="cn">FALSE</span>,
  repos<span class="op">=</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/repositories.html" class="external-link">repositories</a></span><span class="op">(</span><span class="op">)</span>,
  dependencies<span class="op">=</span><span class="cn">TRUE</span>, type<span class="op">=</span><span class="st">"source"</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="citing-the-ramr-package" class="section level3">
<h3 class="hasAnchor">
<a href="#citing-the-ramr-package" class="anchor" aria-hidden="true"></a>Citing the <em><code>ramr</code></em> package</h3>
<p><a href="https://doi.org/10.1101/2020.12.01.403501" class="external-link">Nikolaienko et al., 2020 bioRxiv</a></p>
</div>
<div id="the-data-underlying-ramr-manuscript" class="section level3">
<h3 class="hasAnchor">
<a href="#the-data-underlying-ramr-manuscript" class="anchor" aria-hidden="true"></a>The data underlying <em><code>ramr</code></em> manuscript</h3>
<p><a href="https://doi.org/10.18710/ED8HSD" class="external-link">Replication Data for: “ramr: an R package for detection of rare aberrantly methylated regions”</a></p>
</div>
<div id="ramr-at-bioconductor" class="section level3">
<h3 class="hasAnchor">
<a href="#ramr-at-bioconductor" class="anchor" aria-hidden="true"></a><em><code>ramr</code></em> at Bioconductor</h3>
<p><a href="https://bioconductor.org/packages/release/bioc/html/ramr.html" class="external-link">release</a>, <a href="https://bioconductor.org/packages/devel/bioc/html/ramr.html" class="external-link">development version</a></p>
<hr>
</div>
</div>
</div>
<div id="how-to-use" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-use" class="anchor" aria-hidden="true"></a>How to Use</h1>
<p>Please read package vignettes at <a href="https://bbcg.github.io/ramr/articles/ramr.html" class="external-link">GitHub pages</a> or within the R environment: <code><a href="articles/ramr.html">vignette("ramr", package="ramr")</a></code>, or consult the function’s help pages for the extensive information on usage, parameters and output values.</p>
<p><em><code>ramr</code></em> methods operate on objects of the class <em><code>GRanges</code></em>. The input object for AMR search must in addition contain metadata columns with sample beta values. A typical input object looks like this:</p>
<pre><code>GRanges object with 383788 ranges and 845 metadata columns:
             seqnames    ranges strand |         GSM1235534         GSM1235535         GSM1235536 ...
                &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;numeric&gt;          &lt;numeric&gt;          &lt;numeric&gt; ...
  cg13869341     chr1     15865      * |  0.801634776091808  0.846486905008704   0.86732154737116 ...
  cg24669183     chr1    534242      * |  0.834138820071765  0.861974610731835  0.832557979806823 ...
  cg15560884     chr1    710097      * |  0.711275180750356   0.70461945838556  0.699487225634589 ...
  cg01014490     chr1    714177      * | 0.0769098196182058 0.0569443780518647 0.0623154673389864 ...
  cg17505339     chr1    720865      * |  0.876413362222415  0.885593263385521  0.877944732153869 ...
         ...      ...       ...    ... .                ...                ...                ... ...
  cg05615487    chr22  51176407      * |   0.84904178467798  0.836538383875097   0.81568519870099 ...
  cg22122449    chr22  51176711      * |  0.882444486059592  0.870804215405886  0.859269224277308 ...
  cg08423507    chr22  51177982      * |  0.886406345093286  0.882430879852752  0.887241923657461 ...
  cg19565306    chr22  51222011      * | 0.0719084295670266 0.0845209871264646 0.0689074604483659 ...
  cg09226288    chr22  51225561      * |  0.724145303755024  0.696281176451351  0.711459675603635 ...
</code></pre>
<p>This code shows how to do basic analysis with <em><code>ramr</code></em> using provided data files:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BBCG/ramr" class="external-link">ramr</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ramr</span><span class="op">)</span>

<span class="co"># search for AMRs</span>
<span class="va">amrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getAMR.html">getAMR</a></span><span class="op">(</span><span class="va">ramr.data</span>, <span class="va">ramr.samples</span>, ramr.method<span class="op">=</span><span class="st">"beta"</span>, min.cpgs<span class="op">=</span><span class="fl">5</span>,
               merge.window<span class="op">=</span><span class="fl">1000</span>, qval.cutoff<span class="op">=</span><span class="fl">1e-3</span><span class="op">)</span>

<span class="co"># inspect</span>
<span class="va">amrs</span>
<span class="fu"><a href="reference/plotAMR.html">plotAMR</a></span><span class="op">(</span><span class="va">ramr.data</span>, <span class="va">ramr.samples</span>, <span class="va">amrs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>

<span class="co"># generate the set of all possible genomic regions using sample dataset and</span>
<span class="co"># the same parameters as for AMR search</span>
<span class="va">universe</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getUniverse.html">getUniverse</a></span><span class="op">(</span><span class="va">ramr.data</span>, min.cpgs<span class="op">=</span><span class="fl">5</span>, merge.window<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>

<span class="co"># enrichment analysis of AMRs using R library LOLA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://code.databio.org/LOLA" class="external-link">LOLA</a></span><span class="op">)</span>
<span class="va">hg19.coredb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/LOLA/man/loadRegionDB.html" class="external-link">loadRegionDB</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"LOLACore"</span>, <span class="st">"hg19"</span>, package<span class="op">=</span><span class="st">"LOLA"</span><span class="op">)</span><span class="op">)</span>
<span class="va">core.hits</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/LOLA/man/runLOLA.html" class="external-link">runLOLA</a></span><span class="op">(</span><span class="va">amrs</span>, <span class="va">universe</span>, <span class="va">hg19.coredb</span>, cores<span class="op">=</span><span class="fl">1</span>, redefineUserSets<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The following code generates random AMRs and methylation beta values using provided dataset as a template:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unique random AMRs</span>
<span class="va">amrs.unique</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/simulateAMR.html">simulateAMR</a></span><span class="op">(</span><span class="va">ramr.data</span>, nsamples<span class="op">=</span><span class="fl">10</span>, regions.per.sample<span class="op">=</span><span class="fl">2</span>,
                           min.cpgs<span class="op">=</span><span class="fl">5</span>, merge.window<span class="op">=</span><span class="fl">1000</span>, dbeta<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>

<span class="co"># methylation data with AMRs</span>
<span class="va">data.with.amrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/simulateData.html">simulateData</a></span><span class="op">(</span><span class="va">ramr.data</span>, nsamples<span class="op">=</span><span class="fl">10</span>,
                               amr.ranges<span class="op">=</span><span class="va">amrs.unique</span>, cores<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
  
<span class="co"># that's how regions look like</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"grid.arrange"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="reference/plotAMR.html">plotAMR</a></span><span class="op">(</span><span class="va">data.with.amrs</span>, amr.ranges<span class="op">=</span><span class="va">amrs.unique</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The input (or template) object may be obtained using data from various sources. Here we provide two examples:</p>
<div id="using-data-from-ncbi-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#using-data-from-ncbi-geo" class="anchor" aria-hidden="true"></a>Using data from NCBI GEO</h3>
<p>The following code pulls (NB: very large) raw files from NCBI GEO database, performs normalization and creates <em><code>GRanges</code></em> object for further analysis using <em><code>ramr</code></em> (system requirements: 22GB of disk space, 64GB of RAM)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">minfi</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GEOquery</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span>

<span class="co"># destination for temporary files</span>
<span class="va">dest.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># downloading and unpacking raw IDAT files</span>
<span class="va">suppl.files</span> <span class="op">&lt;-</span> <span class="fu">getGEOSuppFiles</span><span class="op">(</span><span class="st">"GSE51032"</span>, baseDir<span class="op">=</span><span class="va">dest.dir</span>, makeDirectory<span class="op">=</span><span class="cn">FALSE</span>, filter_regex<span class="op">=</span><span class="st">"RAW"</span><span class="op">)</span>
<span class="co"># The default timeout for downloading files in R 4.1 is 60 seconds.</span>
<span class="co"># If code above fails because of that, change your timeout using </span>
<span class="co"># options(timeout=600)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">suppl.files</span><span class="op">)</span>, exdir<span class="op">=</span><span class="va">dest.dir</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">idat.files</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">dest.dir</span>, pattern<span class="op">=</span><span class="st">"idat.gz$"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">idat.files</span>, <span class="va">gunzip</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># reading IDAT files</span>
<span class="va">geo.idat</span> <span class="op">&lt;-</span> <span class="fu">read.metharray.exp</span><span class="op">(</span><span class="va">dest.dir</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geo.idat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(GSM\\d+).*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geo.idat</span><span class="op">)</span><span class="op">)</span>

<span class="co"># processing raw data</span>
<span class="va">genomic.ratio.set</span> <span class="op">&lt;-</span> <span class="fu">preprocessQuantile</span><span class="op">(</span><span class="va">geo.idat</span>, mergeManifest<span class="op">=</span><span class="cn">TRUE</span>, fixOutliers<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># creating the GRanges object with beta values</span>
<span class="va">data.ranges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">genomic.ratio.set</span><span class="op">)</span>
<span class="va">data.betas</span>  <span class="op">&lt;-</span> <span class="fu">getBeta</span><span class="op">(</span><span class="va">genomic.ratio.set</span><span class="op">)</span>
<span class="va">sample.ids</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geo.idat</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">data.ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">data.betas</span>

<span class="co"># data.ranges and sample.ids objects are now ready for AMR search using ramr</span></code></pre></div>
</div>
<div id="using-bismark-cytosine-report-files" class="section level3">
<h3 class="hasAnchor">
<a href="#using-bismark-cytosine-report-files" class="anchor" aria-hidden="true"></a>Using Bismark cytosine report files</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">methylKit</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span>

<span class="co"># file.list is a user-defined character vector with full file names of Bismark cytosine report files</span>
<span class="va">file.list</span>

<span class="co"># sample.ids is a user-defined character vector holding sample names</span>
<span class="va">sample.ids</span>

<span class="co"># methylation context string, defines if the reads covering both strands will be merged</span>
<span class="va">context</span> <span class="op">&lt;-</span> <span class="st">"CpG"</span>

<span class="co"># fitting beta distribution (filtering using ramr.method "beta" or "wbeta") requires</span>
<span class="co"># that most of the beta values are not equal to 0 or 1</span>
<span class="va">min.beta</span> <span class="op">&lt;-</span> <span class="fl">0.001</span>
<span class="va">max.beta</span> <span class="op">&lt;-</span> <span class="fl">0.999</span>

<span class="co"># reading and uniting methylation values</span>
<span class="va">meth.data.raw</span> <span class="op">&lt;-</span> <span class="fu">methRead</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">file.list</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">sample.ids</span><span class="op">)</span>, assembly<span class="op">=</span><span class="st">"hg19"</span>, header<span class="op">=</span><span class="cn">TRUE</span>,
                          context<span class="op">=</span><span class="va">context</span>, resolution<span class="op">=</span><span class="st">"base"</span>, treatment<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sample.ids</span><span class="op">)</span><span class="op">)</span>,
                          pipeline<span class="op">=</span><span class="st">"bismarkCytosineReport"</span><span class="op">)</span>
<span class="va">meth.data.utd</span> <span class="op">&lt;-</span> <span class="fu">unite</span><span class="op">(</span><span class="va">meth.data.raw</span>, destrand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html" class="external-link">isTRUE</a></span><span class="op">(</span><span class="va">context</span><span class="op">==</span><span class="st">"CpG"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># creating the GRanges object with beta values</span>
<span class="va">data.ranges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">meth.data.utd</span><span class="op">)</span>
<span class="va">data.betas</span>  <span class="op">&lt;-</span> <span class="fu">percMethylation</span><span class="op">(</span><span class="va">meth.data.utd</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span>
<span class="va">data.betas</span><span class="op">[</span><span class="va">data.betas</span><span class="op">&lt;</span><span class="va">min.beta</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">min.beta</span>
<span class="va">data.betas</span><span class="op">[</span><span class="va">data.betas</span><span class="op">&gt;</span><span class="va">max.beta</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">max.beta</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">data.ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">data.betas</span>

<span class="co"># data.ranges and sample.ids objects are now ready for AMR search using ramr</span></code></pre></div>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor" aria-hidden="true"></a>License</h2>
<p>Artistic License/GPL</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/ramr">https://​www.bioconductor.org/​packages/​ramr</a>
</li>
<li>Browse source code at <br><a href="https://github.com/BBCG/ramr/">https://​github.com/​BBCG/​ramr/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/BBCG/ramr/issues">https://​github.com/​BBCG/​ramr/​issues</a>
</li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>


<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Oleksii Nikolaienko <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5910-4934" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/BBCG/ramr/actions" class="external-link"><img src="https://github.com/BBCG/ramr/workflows/R-CMD-check-bioc/badge.svg"></a></li>
<li><a href="https://codecov.io/gh/BBCG/ramr" class="external-link"><img src="https://codecov.io/gh/BBCG/ramr/branch/master/graph/badge.svg"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/ramr.html" class="external-link"><img src="https://bioconductor.org/shields/years-in-bioc/ramr.svg"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Oleksii Nikolaienko.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
